orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/strategies/RollingRedBlackStrategy.groovy

```diff
@@ -122,6 +114,14 @@ class RollingRedBlackStrategy implements Strategy, ApplicationContextAware {
 
       Stage parentCreateServerGroupStage = stage.directAncestors().find() { it.type == CreateServerGroupStage.PIPELINE_CONFIG_TYPE }
 
+      if (parentCreateServerGroupStage.status == ExecutionStatus.NOT_STARTED) {
+        // No point in composing the flow if we are called to plan "beforeStages" since we don't have any STAGE_BEFOREs.
```

Comment:
I don't see an explicit connection between the condition being checked and being called to plan beforeStages, can you elaborate?

Reply:
Remember this one: https://github.com/spinnaker/orca/blob/master/orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/StageDefinitionBuilder.java#L83

Reply:
what happens is:
`DeployStrategyStage` implemented `aroundStages` which calls our `composeFlow`
the `DeployStrategyStage::aroundStages` is called from `StageDefinitionBuilder::beforeStages` (java) which is called from `StageDefinitionBuilder::buildBeforeStages` (kotlin), which is called from `RunStageHandler`